### PR TITLE
feat: migrate to pnpm

### PR DESCRIPTION
**Pull request by QuiiBz**
Migrate from `yarn` to `pnpm`

### CODE REVIEW

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -12,24 +12,30 @@ jobs:
     runs-on: ubuntu-20.04
     steps:
       - uses: actions/checkout@v2.3.5
+      - uses: pnpm/action-setup@646cdf48217256a3d0b80361c5a50727664284f2
```
Comment (adriengibrat) [inline .github/workflows/ci_cd.yml:15]:
```suggestion
      - uses: pnpm/action-setup@v2.2.1
```

File: .husky/commit-msg [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
-yarn run commitlint --edit $1
+pnpm commitlint --edit $1
```
Comment (adriengibrat) [inline .husky/commit-msg:4]:
```suggestion
pnpm commitlint -- --edit $1
```
i'm not sure, but i guess ?

File: .github/workflows/ci_cd.yml (hunk 1 above)
Comment (chambo-e) [inline .github/workflows/ci_cd.yml:15]: Can we use a stable version ?

File: .husky/commit-msg (hunk 2 above)
Comment (chambo-e) [inline .husky/commit-msg:4]:
```suggestion
pnpm run commitlint --edit $1
```

?

File: README.md [hunk 3]
```diff
@@ -7,9 +7,7 @@ This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next
 First, run the development server:
 
 ```bash
-npm run dev
-# or
-yarn dev
+pnpm dev
```
Comment (chambo-e) [inline README.md:10]:
```suggestion
pnpm run dev
```

File: src/pages/home/GettingStarted.tsx [hunk 4]
```diff
@@ -4,8 +4,8 @@ import React from 'react'
 import CopyBox from '../../components/CopyBoxCommand'
 import GithubAndDocumentationButtons from '../../components/GithubAndDocumentationButtons'
 
-const install = `yarn add @scaleway/ui @emotion/react @emotion/styled`
-const installNpm = `npm i @scaleway/ui @emotion/react @emotion/styled`
```
Comment (adriengibrat) [inline src/pages/home/GettingStarted.tsx:8]: RIP npm :crying_cat_face:

File: .github/workflows/ci_cd.yml (hunk 1 above)
Comment (QuiiBz) [reply .github/workflows/ci_cd.yml:15]: This is the version used in the docs of actions/setup-node: https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-dependencies

**Review by matthprost — APPROVED** [review]

File: .husky/commit-msg (hunk 2 above)
Comment (QuiiBz) [reply .husky/commit-msg:4]: Only necessary when running scripts, not executing packages

File: src/pages/home/GettingStarted.tsx (hunk 4 above)
Comment (adriengibrat) [reply src/pages/home/GettingStarted.tsx:8]: We should add, not replace in the public doc:
image

Comment (adriengibrat) [reply src/pages/home/GettingStarted.tsx:8]: We should add, not replace in the public doc:
image

Comment (adriengibrat) [reply src/pages/home/GettingStarted.tsx:8]: We should add, not replace in the public doc:
image

File: .husky/commit-msg (hunk 2 above)
Comment (QuiiBz) [reply .husky/commit-msg:4]: `commitlint` is not a script but a package, so it doesn't make any sense. Instead we can use `exec`:
```
pnpm exec commitlint --edit $1
```

File: .github/workflows/ci_cd.yml (hunk 1 above)
Comment (chambo-e) [reply .github/workflows/ci_cd.yml:15]: It seems that it's a ref inside the main branch https://github.com/pnpm/action-setup/tree/646cdf48217256a3d0b80361c5a50727664284f2

Can we try with v3 ?

Comment (adriengibrat) [reply .github/workflows/ci_cd.yml:15]: yep, but it an old version https://github.com/pnpm/action-setup/releases/tag/v2.0.1 from 2021 ;)

better to use a tag release than a commit  and using the latest would be better:
https://github.com/pnpm/action-setup/releases/tag/v2.2.1

**Review by adriengibrat — APPROVED** [review]

**Review by chambo-e — APPROVED** [review]

**Pull request merged**